src/main/java/org/example/Operations/QueryOperations.java

```diff
@@ -65,12 +71,18 @@ public Response addSpecificQuery(String token, String name, String sql) {
         Map<Object, Object> formData = new HashMap<>();
         formData.put("op", "add_specific_query");
         formData.put("name", name);
-        formData.put("sql", sql);
+
+        // need to URL encode the spaces in the query to prevent illegal characters in the query
+        try {
+            String encodedSql = URLEncoder.encode(sql, StandardCharsets.UTF_8.toString());
+            formData.put("sql", encodedSql);
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
```

Comment:
Consider the following:
- The client is running in an environment with no device for stdout/stderr
- `UnsupportedEncodingException` is thrown

What happens to the output of `e.printStackTrace()`?
Will the developer or user have enough information to figure out where things went wrong?

Reply:
Logging could help, but that doesn't give the developer a chance to react to the exception.

If you let the exception be thrown, the developer can catch it and make decisions based on their own requirements.

Reply:
Done.